src/main/java/org/terasology/launcher/gui/javafx/Dialogs.java

```diff
@@ -31,17 +35,38 @@
 import java.nio.file.Path;
 import java.util.Optional;
 import java.util.concurrent.ExecutionException;
-import java.util.concurrent.FutureTask;
+import java.util.function.Supplier;
 
-public final class GuiUtils {
+public final class Dialogs {
 
-    private static final Logger logger = LoggerFactory.getLogger(GuiUtils.class);
+    private static final Logger logger = LoggerFactory.getLogger(Dialogs.class);
 
-    private GuiUtils() {
+    private Dialogs() {
+    }
+
+    private static <T> T runOnEventThread(Supplier<T> producer) {
+        T result = null;
+        if (Platform.isFxApplicationThread()) {
+            result = producer.get();
```

Comment:
I wondered: What would happen if this `if` clause weren't here? But I have a horrible suspicion about what the answer is: we'd ask it to "run later", and then move on to the blocking `task.get()`, and then "later" would never come because we're blocking the application thread.

ughhhhh.

Reply:
I was wondering about that, too. At some point, we've put this `if` construct in the main code, and I hope for some reason (maybe the one you described above) 🙈  I did not double-check that, though, and this is probably something we could unit test in the future, too.